include/gmsh.tcc

```diff
@@ -138,13 +143,25 @@ void GMSH::compute_stresses() {
 
   //! Loop through the points to get vertical and horizontal stresses
   //! Note that tau (shear stress) is assumed 0
-  for (const auto& point : vertices_) {
-    ver_stress =
-        conv_factor * (-(max_height - point->coordinates().at(2))) * density;
+  for (const auto& point : coords_) {
+    ver_stress = conv_factor * (-(max_height - point.at(2))) * density;
     hor_stress = ver_stress * k0;
     std::array<double, 6> stress{hor_stress, hor_stress, ver_stress, 0, 0, 0};
-    stress_.emplace_back(stress);
+    stress_.push_back(stress);
+  }
+
+  std::cout << "Computed initial stresses.\n";
+}
+
+//! \brief generate material points
+void GMSH::generate_points() {
+
+  const unsigned ndimension = 3;
+
+  for (unsigned i = 0; i < nvertices_; ++i) {
+    vertices_.emplace_back(
```

Comment:
Yes! I designed the constructor of `points` to contain `id`, `coord`, and `stress`.